### PR TITLE
Phase plot showhide colorbar

### LINKED ISSUE #1736: PhasePlot is missing [show,hide]_axes() and [show,hide]_colorbar()
Currently the `PlotWindow` plots all have `show_axes`, `hide_axes`, `show_colorbar`, and `hide_colorbar` implemented. Since the actual functionality for hiding and displaying the colorbar is handled by the lower-level matplotlib wrapper objects (e.g. `PlotMPL`, `WindowPlotMPL`, and `PhasePlotMPL`) the implementations of these functions on the `PlotWindow` class are pretty simple:

https://github.com/yt-project/yt/blob/51cf1ce0825dd1e9ef6bbbf8e9a83a82a768cd7e/yt/visualization/plot_window.py#L1046-L1051

For example, I've pasted the implementation of `hide_colorbar` above.

As pointed out by Stephannie Tonneson on the mailing list, currently the `PhasePlot` object does not have these methods. This is a bit of an API inconsistency. I think it should be possible to move these methods to the base `PlotContainer` class or barring that, add separate implementations for these function on the `PhasePlot` class itself. This should be simple since the underlying `PhasePlotMPL` class already can handle hiding the axes and colorbar:

```python
plot = yt.PhasePlot(ds.all_data(), 'density', 'temperature', 'cell_mass')
p = plot.plots['cell_mass']
p.hide_colorbar()
plot.save()
```

### PR DESCRIPTION
**Pull request by opaco**
Moved show/hide_colorbar and show/hide_axes from PlotWindow to PlotContainer. Added property function in profile_plotter.py so that PlotWindow implementation of these functions works with ProfilePlot.

Added note in yt document that PhasePlot has similar attributes, such as show and hide colorbar, as SlicePlot.

Resolves #1736 

## PR Checklist

- [ ] Code passes flake8 checker
- [ ] New features are documented, with docstrings and narrative docs
- [ ] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

**Comment by ngoldbaum** [comment]
The test failures are unrelated to this PR and come from my clearing the travis cache recently (I think). I will fix it separately, sorry for the trouble!

File: yt/visualization/plot_container.py [hunk 1]
```diff
@@ -651,6 +651,139 @@ def _get_axes_unit_labels(self, unit_x, unit_y):
...
+            object.
+
+        Examples
+        --------
+
+        This will save an image with no colorbar.
+
+        >>> import yt
+        >>> ds = yt.load('IsolatedGalaxy/galaxy0030/galaxy0030')
+        >>> s = SlicePlot(ds, 2, 'density', 'c', (20, 'kpc'))
```
Comment (MatthewTurk) [inline yt/visualization/plot_container.py]: I think this, and the one below, need to be updated to be `PhasePlot` instances

File: yt/visualization/plot_container.py [hunk 2]
```diff
@@ -651,6 +651,139 @@ def _get_axes_unit_labels(self, unit_x, unit_y):
...
+            See note below for more details.
+
+        Examples
+        --------
+
+        This will save an image with no axes.
+
+        >>> import yt
+        >>> ds = yt.load('IsolatedGalaxy/galaxy0030/galaxy0030')
+        >>> s = SlicePlot(ds, 2, 'density', 'c', (20, 'kpc'))
```
Comment (MatthewTurk) [inline yt/visualization/plot_container.py]: same here

**Comment by ngoldbaum** [comment]
I guess in principle we could edit the docstrings for each class but that's overkill for this PR. These functions are still attached to `AxisAlignedSlicePlot` instances, but now additionally they're attached to `PhasePlot` instances.

**Comment by ngoldbaum** [comment]
Congratulations on your first yt contributon! :100: 🥇 🎆 🎉

**Pull request merged**